components/locale_core/src/preferences/extensions/unicode/keywords/numbering_system.rs

```diff
@@ -2,22 +2,209 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-use crate::preferences::extensions::unicode::errors::PreferencesParseError;
-use crate::preferences::extensions::unicode::struct_keyword;
-use crate::{extensions::unicode::Value, subtags::Subtag};
+use crate::preferences::extensions::unicode::enum_keyword;
 
-struct_keyword!(
-    /// TODO
-    NumberingSystem,
-    "nu",
-    Subtag,
-    |input: Value| {
-        input
-            .into_single_subtag()
-            .map(Self)
-            .ok_or(PreferencesParseError::InvalidKeywordValue)
-    },
-    |input: NumberingSystem| {
-        crate::extensions::unicode::Value::from_subtag(Some(input.0))
-    }
-);
+enum_keyword!(
+    /// A Unicode Number System Identifier defines a type of number system.
+    ///
+    /// The valid values are listed in [LDML](https://unicode.org/reports/tr35/#UnicodeNumberSystemIdentifier).
```

Comment:
Issue: Why did you change this to a closed enum? This seems worse than an open enum because:

1. This list changes frequently and it's silly to update it here. We have to do that every time in ECMA-402 and it doesn't solve any real-world problems.
2. We have to go back and forth between a TinyAsciiStr and an enum discriminant. It would be nice to just reference the string, which we can do with `DataMarkerAttributes::try_from_str`.

Reply:
My logic was that numbering systems are actually quite stable and don't change frequently (like, currencies or timezones). I am surprised that you say that they change often - I'd expect the list to only grow, and not often (generating a new numbering system is non trivial).

My DX reason was that writing `NumberingSystem::Latn` is nicer than `NumberingSystem::from(value!("latn"))` and this is something I can see developers actually writing, or listing in Preferences.

My position is weak here and I can switch it if you prefer an open struct after my explanation.

Reply:
Numbering systems have been added every year for the last 4 years: https://github.com/tc39/ecma402/issues/884

You shouldn't need to update ICU4X to be able to consume data that reflects the new numbering systems.

I don't buy the ergonomics argument. Just have `NumberingSystem::latin()` which you can write as a const fn or even `NumberingSystem::LATIN` which you can write as an associated constant.